pkg/graveler/ref/manager.go

```diff
@@ -47,23 +47,75 @@ func NewKVRefManager(executor batch.Batcher, kvStore kv.StoreMessage, db db.Data
 }
 
 func (m *KVManager) GetRepository(ctx context.Context, repositoryID graveler.RepositoryID) (*graveler.Repository, error) {
-	return m.db.GetRepository(ctx, repositoryID)
+	repo := graveler.RepositoryData{}
+	_, err := m.kvStore.GetMsg(ctx, graveler.RepoPartition(), []byte(graveler.RepoPath(repositoryID)), &repo)
+	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrRepositoryNotFound
+		}
+		return nil, err
+	}
+	return graveler.RepoFromProto(&repo).Repository, nil
+}
+
+func (m *KVManager) createBareRepository(ctx context.Context, repositoryID graveler.RepositoryID, repository graveler.Repository) error {
```

Comment:
Seems like we don't really need this internal method anymore - we can just use the public interface

Reply:
I'm keeping this on hold at the moment, as I would like to discuss a possible solution with `repositoryStatus`
If that solution is a go, we would still need the internal `createBareRepository`, as the public `CreateBareRepository` will set the repository status right after creation, unlike `CreateRepository`